sam.c

```diff
@@ -230,7 +230,7 @@ int bam_hdr_write(BGZF *fp, const bam_hdr_t *h)
     char buf[4];
     int32_t i, name_len, x;
     // write "BAM1"
-    strncpy(buf, "BAM\1", 4);
+    memcpy(buf, "BAM\1", 4);
```

Comment:
Ugh. If you're going to touch this piece of code anyway, why not actually improve it by nuking `buf` entirely:
`if (bgzf_write(fp, "BAM\1", 4) …`

Reply:
Good point...